### DISCUSSION TITLE
Source code vs cache

### DISCUSSION THREAD

**Discussion by itacdonev** [executablebooks/meta#258] [Q&A]
Hello,

I am working on a larger project where the documentation for the project is built using the jupyter-book. It's a very efficient way to communicate the projects steps and the results with other colleagues. 
I have a folder structure as follows:
```
|──input
|──techdoc         <- juypter book contents
   └── content     <- ipynb files
|──src             <- source code (*.py files)
```

I am using `cache` for notebooks build with `execute_notebooks   : cache` due to large project scope, as it takes considerably less time.

**Question**
I have a problem with the jupyter-book build when changing the source code and nothing else in the notebook. 

The build does not recognize the change, since there are no cells being physically changed in the notebook and it pulls the data from the cache for the build. 

When I run the cells in the notebook which uses the functions from the updated source code the outputs of the cells are updated. The notebook uses 
```
%reload_ext autoreload
%autoreload 2
```
**Is there a way for the build to recognize the change in the .py file in the `src/` folder not to trigger the cache for that notebook when build is triggered when there are no changes in the cells per se in that specific notebook?**

**Comment by chrisjsewell** [comment]
Hi @itacdonev no there is not currently a way to achieve this I'm afraid. But I think this could certainly be achieved and will add it as an improvement issue. Something like, adding a mapping of folders/files to notebooks that should be "invalidated"if they change

**Reply by chrisjsewell** [reply]
see https://github.com/executablebooks/MyST-NB/issues/299

**Reply by itacdonev** [reply]
Great! Thanks! 
This will help a lot, especially for larger projects.
